### DISCUSSION TITLE
Sky130-OpenRoad Tutorial Place and Route Fail

### DISCUSSION THREAD

**Discussion by johnw08** [ucb-bar/hammer#857] [Support (Q&A)] [answered]
Hello,
I'm following the Sky130-OopenRoad tutorial and I was able to get it work until the Place and Route part. After running "make par tutorial=sky130-openroad" command I got the following error message:

```
[par] [Exec ...d/bin/openroad -no_init -log /...] create_clock clock_uncore_clock -name clock_uncore_clock -period 50.0
[par] [Exec ...d/bin/openroad -no_init -log /...] [ERROR STA-0489] pin 'clock_uncore_clock' not found.
[par] [Exec ...d/bin/openroad -no_init -log /...] Error: clock_constraints_fragment.sdc, 1 STA-0489
[par] ERROR: OpenROAD returned with a nonzero exit code: 1.
```

Please help. Thank you!

**Comment by johnw08** [comment, marked as answer]
I found out the problem. The clock name should be "clock_uncore" instead of "clock_uncore_clock" in all the yml config files. Now I ran into a different error. 

```
[par] [Exec ...d/bin/openroad -no_init -log /...] (hammer) detailed_placement
[par] [Exec ...d/bin/openroad -no_init -log /...] [INFO DPL-0034] Detailed placement failed on the following 2 instances:
[par] [Exec ...d/bin/openroad -no_init -log /...] [INFO DPL-0035]  system/tile_prci_domain/element_reset_domain_rockettile/frontend/icache/data_arrays_0_0/data_arrays_0_0_ext/mem_0_0
[par] [Exec ...d/bin/openroad -no_init -log /...] [INFO DPL-0035]  system/tile_prci_domain/element_reset_domain_rockettile/frontend/icache/tag_array_0/tag_array_0_ext/mem_0_0
[par] [Exec ...d/bin/openroad -no_init -log /...] [ERROR DPL-0036] Detailed placement failed.
[par] [Exec ...d/bin/openroad -no_init -log /...] Error: par.tcl, 299 DPL-0036
```

**Comment by arunlee77** [comment]
@johnw08  did you get any solution for this issue. 

My observation is that, the example documentation given in https://chipyard.readthedocs.io/en/stable/VLSI/Sky130-OpenROAD-Tutorial.html is completely broken and will not work. But it seems no one has time or interest to fix it. 

My understanding is that the openroad used in the above link is not compatible with the hammer-vlsi which gets installed as part of the initial setup. 
I tried experimenting with latest openroad, but could not complete the tutorial. 

my current status is that i am able to run   `make par tutorial=sky130-openroad ` but it get stuck at 

```
[par] [Exec ...l/bin/openroad -no_init -log /...] (hammer) buffer_ports
[par] [Exec ...l/bin/openroad -no_init -log /...] [INFO RSZ-0027] Inserted 41 input buffers.
[par] [Exec ...l/bin/openroad -no_init -log /...] [INFO RSZ-0028] Inserted 37 output buffers.
[par] [Exec ...l/bin/openroad -no_init -log /...] (hammer) repair_design
[par] [Exec ...l/bin/openroad -no_init -log /...] [INFO RSZ-0058] Using max wire length 3431um.

```
